### DISCUSSION TITLE
Adding bulk points for a Voronoi mesh

### DISCUSSION THREAD

**Discussion by CNOT** [DanielVandH/DelaunayTriangulation.jl#209] [General]
Let's say I figured a way to generate the cell centers for a Voronoi mesh near the sharp features such as concave corners in my object (In my case I used VoroCrust to generate these generator seeds for a complex boundary). Is there a way to add the rest of the Voronoi seeds, let's say with a structured mesh? 
![Screenshot From 2024-10-31 02-42-07](https://github.com/user-attachments/assets/5a8786ce-0646-47ce-9209-942f941d71f4)

**Comment by DanielVandH** [comment]
I don't think I have implemented `add_point!` yet for tessellations, if I understand correctly that you want to update the tessellation after a new generator is added?
It wouldn't be too hard to implement though since it's already implemented for triangulations.

**Reply by CNOT** [reply]
Hi @DanielVandH 
Any updates on whether this has been implemented yet or if it's upcoming? It would be also helpful if I could just give all the generator seeds and get the Voronoi mesh, like what Makie's `voronoiplot` does, however the outcome being a `VoronoiTesselation` instead of a FigureAxisPlot.

**Reply by DanielVandH** [reply]
Not sure when I'll get to it, sorry. I'm very busy these days and don't have the time to spend on development like I used to sadly.
I'm happy to help with a PR implementing it though.

**Reply by DanielVandH** [reply]
Also for the last point about just making a voronoi tessellation from points directly, I think

function voronoi(points; kwargs....)
    tri = triangulate(points; kwargs...)
    voronoi(tri)
end

Would work, although you need to add the other keyword arguments (separate to kwargs...) that voronoi uses. If you want to add that I'm happy to accept it

**Reply by CNOT** [reply]
Thanks for the workaround. I'm excruciatingly busy too these days, but I may try working on the PR if I find the free time.
